datalad/tests/test_classes.py

```diff
@@ -0,0 +1,56 @@
+"""
+Tests to test implementation of basic classes for datalad.
+
+Note: There's not a lot to test by now.
+"""
+__author__ = 'Benjamin Poldrack'
+
+import os.path
+from shutil import rmtree
+
+from nose.tools import assert_raises, assert_is_instance, assert_true
+from git.exc import GitCommandError
+
+from datalad.support.gitrepo import GitRepo
+from datalad.support.annexrepo import AnnexRepo
+from datalad.support.dataset import Dataset
+
+def test_GitRepo():
+
+    pathToTestRepo = os.path.expanduser('~/test_git_repo')
+    gr = GitRepo(pathToTestRepo, 'http://psydata.ovgu.de/forrest_gump/.git')
```

Comment:
After thinking a bit more, reply on this topic at datalad-team, please ;-)

2015-02-23 17:22 GMT+01:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/tests/test_classes.py
> <https://github.com/datalad/datalad/pull/42#discussion_r25176952>:
>
> > +__author__ = 'Benjamin Poldrack'
> > +
> > +import os.path
> > +from shutil import rmtree
> > +
> > +from nose.tools import assert_raises, assert_is_instance, assert_true
> > +from git.exc import GitCommandError
> > +
> > +from datalad.support.gitrepo import GitRepo
> > +from datalad.support.annexrepo import AnnexRepo
> > +from datalad.support.dataset import Dataset
> > +
> > +def test_GitRepo():
> > +
> > +    pathToTestRepo = os.path.expanduser('~/test_git_repo')
> > +    gr = GitRepo(pathToTestRepo, 'http://psydata.ovgu.de/forrest_gump/.git')
>
> definetly not in the ~ -- use with_tempfile and see how it is used in
> other tests
>
> I am not sure if these tests should depend on any external existing repo
> ATM, and definitely not big mighty forrest_gump -- for that we have
> testkraut tests atm.
>
> I would say -- we need a set of test repos available as git submodules
> e.g. under datalad/tests/testrepos. And then these tests would
> parameterized by running tests across those available. This way we could
> generate for different versions of git-annex etc.... For now -- leave as
> is, I will think a bit more on how we should do it without incurring
> unneeded burden
>
> Just create/use one. Use @with_tree
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datalad/datalad/pull/42/files#r25176952>.
>